pilot/pkg/model/jwks_resolver.go

```diff
@@ -238,29 +244,28 @@ func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string) (string, erro
 	}
 
 	var err error
-	var pubKey string
+
 	if jwksURI == "" {
 		// Fetch the jwks URI if it is not hardcoded on config.
 		jwksURI, err = r.resolveJwksURIUsingOpenID(issuer)
 	}
 	if err != nil {
 		log.Errorf("Failed to jwks URI from %q: %v", issuer, err)
 	} else {
-		var resp []byte
-		resp, err = r.getRemoteContentWithRetry(jwksURI, networkFetchRetryCountOnMainFlow)
-		if err != nil {
-			log.Errorf("Failed to fetch public key from %q: %v", jwksURI, err)
-		}
-		pubKey = string(resp)
-	}
+		// reset the ticker to hit jwksuri more often from istiod
+		key.jwksURI = jwksURI
+		r.refreshTicker.Reset(time.Second)
+		r.refreshIntervalOnFailure = time.Second
+		r.tickerCounter = 5
 
+	}
 	r.keyEntries.Store(key, jwtPubKeyEntry{
```

Comment:
had to pass empty public key because we can't wait for the cache to be updated, if we are waiting for it then there is no difference in making request to jwksuri in the main flow.